core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -7302,7 +7302,9 @@ QueryModel optimise(
             moveTimestampToChooseModel(rewrittenModel);
             propagateTopDownColumns(rewrittenModel, rewrittenModel.allowsColumnsChange());
             rewriteMultipleTermLimitedOrderByPart2(rewrittenModel);
-            authorizeColumnAccess(sqlExecutionContext, rewrittenModel);
+            if (!sqlExecutionContext.isValidationOnly()) {
+                authorizeColumnAccess(sqlExecutionContext, rewrittenModel);
```

Comment:
Shouldn't we keep this check even in validation-only mode? Otherwise, a user who shouldn't "see" some of the columns, will be able to learn that they exist and their types by validating a query.